server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -532,7 +542,9 @@ public void messageReceived(FieldCapabilitiesNodeRequest request, TransportChann
                                 unmatched.clear();
                                 failures.clear();
```

Comment:
Good catch! I updated it to clear unmatched and failures only if we include fields with no value (as it was before, since we really care for just the first response from an index) and keep them if we exclude empty fields.